broker/data-access-layer/iaas/src/AwsClient.js

```diff
@@ -0,0 +1,308 @@
+'use strict';
+
+const _ = require('lodash');
+const Promise = require('bluebird');
+const S3 = require('aws-sdk/clients/s3');
+const logger = require('@sf/logger');
+const {
+  CONST,
+  errors: {
+    NotFound,
+    Unauthorized
+  },
+  commonFunctions: {
+    streamToPromise
+  }
+} = require('@sf/common-utils');
+
+const ComputeClient = require('./ComputeClient');
+const BaseCloudClient = require('./BaseCloudClient');
+
+class AwsClient extends BaseCloudClient {
+  constructor(settings) {
+    super(settings);
+    this.constructor.validateParams(this.settings);
+    this.storage = this.constructor.createStorageClient(_
+      .chain(this.settings)
+      .omit('name')
+      .set('provider', this.provider)
+      .value()
+    );
+    this.blockstorage = this.constructor.createComputeClient(_
+      .chain(this.settings)
+      .set('provider', this.provider)
+      .value()
+    );
+    this.storage.listObjectsAsync = Promise.promisify(this.storage.listObjects, {});
+    this.storage.deleteObjectAsync = Promise.promisify(this.storage.deleteObject, {});
+    this.storage.uploadAsync = Promise.promisify(this.storage.upload, {});
+  }
+
+  createDiskFromSnapshot(snapshotId, zones, options = {}) {
```

Comment:
Involves operation with respect to `computeClient`, so directly took from `CloudProviderClient.js`